Source/Utils/Average.cpp

```diff
@@ -3,27 +3,109 @@
 using namespace amrex;
 
 void
-Average::ToCellCenter ( MultiFab& mf_out,
-                        const MultiFab& mf_in,
-                        const int dcomp,
-                        const int ngrow,
-                        const int scomp,
-                        const int ncomp )
+Average::CoarsenAndInterpolateLoop ( MultiFab& mf_dst,
+                                     const MultiFab& mf_src,
+                                     const int dcomp,
+                                     const int scomp,
+                                     const int ncomp,
+                                     const int ngrow,
+                                     const IntVect crse_ratio )
 {
+    // Staggering of source fine MultiFab and destination coarse MultiFab
+    const IntVect stag_src = mf_src.boxArray().ixType().toIntVect();
+    const IntVect stag_dst = mf_dst.boxArray().ixType().toIntVect();
+
+    AMREX_ALWAYS_ASSERT_WITH_MESSAGE( mf_src.nGrowVect() >= IntVect(stag_dst-stag_src),
```

Comment:
```
    IntVect(stag_dst-stag_src),
```
Isn't the difference between two IntVect already an IntVect?

Reply:
Actually I think this check should include the number of guard cells required too...
Is there an example where we need to update the guard cells? Otherwise, I am fine with `AMREX_ALWAYS_ASSERT(ngrow==0)`, as it may be difficult to have the right number of guard cells, so maybe we would do differently (update valid cells only, and then exchanged guard cells on the coarsened MultiFab).

Reply:
Yes, the compiler does not complain if I leave simply `stag_dst-stag_src`, so I can modify the code accordingly. Regarding the guard cells, I have encountered problems when I do not fill the guard cells (in the equivalent previous calls to `Average::ToCellCenter`) in tests using back-transformed diagnostics (not yet sure of the reason, though).